aws-datastore/src/test/java/com/amplifyframework/datastore/AWSDataStorePluginTest.java

```diff
@@ -542,6 +550,89 @@ private static ApiCategory mockApiPluginWithExceptions() throws AmplifyException
+        final CountDownLatch latch = new CountDownLatch(1);
+        Person expectedResult = createPerson("Test", "Dummy I");
+        final Person[] actualResult = {null};
+        Consumer<DataStoreItemChange<Person>> onObserveResult = spy(new Consumer<DataStoreItemChange<Person>>() {
+            @Override
+            public void accept(@NonNull DataStoreItemChange<Person> value) {
+                latch.countDown();
+                actualResult[0] = value.item();
+            }
+        });
+        awsDataStorePlugin.observe(Person.class,
+            QueryField.field("type").eq(StorageItemChange.Type.CREATE),
+            value -> { },
+            onObserveResult,
+            error -> {
+                LOG.error("Error: " + error);
+            },
+            () -> { }
+        );
+        synchronousDataStore.save(expectedResult);
+        latch.await(TIMEOUT_MS, TimeUnit.SECONDS);
+        verify(onObserveResult).accept(any());
+        assertEquals(actualResult[0], expectedResult);
+    }
+
+    /**
+     * Verify that the observe api is not invoke when the item changed does not match the predicate.
+     *
+     * @throws JSONException        on failure to arrange plugin config.
+     * @throws AmplifyException     on failure to arrange API plugin via Amplify facade.
+     * @throws InterruptedException If interrupted while test observer awaits terminal result.
+     */
+    @Test
+    public void observeWithoutMatchingPredicate() throws InterruptedException, AmplifyException, JSONException {
+        AWSDataStorePlugin awsDataStorePlugin = AWSDataStorePlugin.builder()
+                .modelProvider(modelProvider)
+                .build();
+        JSONObject dataStorePluginJson = new JSONObject()
+                .put("syncIntervalInMinutes", 60);
+        awsDataStorePlugin.configure(dataStorePluginJson, context);
+        awsDataStorePlugin.initialize(context);
+        Amplify.Hub.publish(HubChannel.DATASTORE, HubEvent.create(InitializationStatus.SUCCEEDED));
+        SynchronousDataStore synchronousDataStore = SynchronousDataStore.delegatingTo(awsDataStorePlugin);
+        Person expectedResult = createPerson("Test", "Dummy I");
+        final Person[] actualResult = {null};
+        Consumer<DataStoreItemChange<Person>> onObserveResult = spy(new Consumer<DataStoreItemChange<Person>>() {
+            @Override
+            public void accept(@NonNull DataStoreItemChange<Person> value) {
+                actualResult[0] = value.item();
+            }
+        });
+        awsDataStorePlugin.observe(Person.class,
+            QueryField.field("type").eq(StorageItemChange.Type.UPDATE),
+            value -> { },
+            onObserveResult,
+            error -> {
+                LOG.error("Error: " + error);
+            },
+            () -> { }
+        );
+        synchronousDataStore.save(expectedResult);
+        Thread.sleep(1000L);
```

Comment:
Hmm, this sleep isn't necessary because `synchronousDataStore` blocks on each operation (unless you tested it and it failed without the sleep). We need to shave off unnecessary waits whenever possible in tests to reduce the overall integration test runtime